index.js

```diff
@@ -1,3 +1,13 @@
 #!/usr/bin/env node
 process.chdir(__dirname);
+
+// Perform node version check before loading any other files or modules
+// Doing this check as soon as possible allows us to avoid ES6 parser errors or other issues
+var pkg = require("./package.json");
+if (!require("semver").satisfies(process.version, pkg.engines.node)) {
+	console.error("=== WARNING!");
```

Comment:
Is it worth using your logger instead, considering we control its content (and can make sure it works on Node.js < 4)?